api/insight/socket.io.go

```diff
@@ -100,54 +102,71 @@ func NewSocketServer(params *chaincfg.Params, txGetter txhelpers.RawTransactionG
 		PingTimeout:  5 * time.Second,
 		Transports:   []transport.Transport{wsTrans},
 	}
-	server, err := socketio.NewServer(opts)
+	socketIOServer, err := socketio.NewServer(opts)
 	if err != nil {
 		apiLog.Errorf("Could not create socket.io server: %v", err)
 		return nil, err
 	}
 
-	// Set maximum number of connections.
-	//server.SetMaxConnection(16384)
-
 	// Each address subscription uses its own room, which has the same name as
 	// the address. The number of subscribers for each room is tracked.
-	addrs := roomSubscriptionCounter{
-		RWMutex: new(sync.RWMutex),
-		c:       make(map[string]int),
+	addrs := &roomSubscriptionCounter{
+		c: make(map[string]int),
+	}
+
+	server := &SocketServer{
+		Server:           socketIOServer,
+		params:           params,
+		watchedAddresses: addrs,
+		txGetter:         txGetter,
 	}
 
 	server.OnConnect("/", func(so socketio.Conn) error {
 		// New connections automatically join the inv and sync rooms.
 		so.Join("inv")
 		so.Join("sync")
-		apiLog.Debugf("New socket.io connection. %d clients are connected.",
-			server.RoomLen("inv"))
+		//so.SetContext(uint32(0))
+		apiLog.Debugf("New socket.io connection (%s). %d clients are connected.",
+			so.ID(), server.RoomLen("inv"))
 		return nil
 	})
 
 	// Subscription to a room checks the room name is as expected for an
 	// address, joins the room, and increments the room's subscriber count.
-	server.OnEvent("/", "subscribe", func(so socketio.Conn, room string) {
+	server.OnEvent("/", "subscribe", func(so socketio.Conn, room string) string {
+		return "i got nothin"
 		if len(room) > 64 || !isAlphaNumeric(room) {
-			return
+			return "bad address"
 		}
 		if _, err := dcrutil.DecodeAddress(room, params); err == nil {
```

Comment:
Ah, I was shadowing `err` here, but using it below in `return "bad address: " + err.Error()` where it was unexpectedly nil.

At least all this debugging revealed that other third-party clients are manually subscribing to "inv".  I suspect we should remove `so.Join("inv")` from the `OnConnect` handler.